src/object/cli_ec.c

```diff
@@ -367,25 +475,48 @@ ec_get_tgt_set(daos_iod_t *iods, unsigned int nr, struct daos_oclass_attr *oca,
 						iods[i].iod_size + rs - 1;
 
 			/* No partial-parity updates, so this function won't be
-			 * called if parity is present.
+			 * called if parity is present in a update request.
+			 * For fetch (!parity_include), the code allows parity
+			 * segments to be requested (and handles them
+			 * separately).
+			 */
+			if (PARITY_INDICATOR & rs) {
+				/* This allows selecting a parity target
+				 * for fetch. If combined with regualar
+				 * data extents, parity ranges must come
+				 * first in the the recx array.
+				 */
+				D_INFO("rs: %lu\n", rs);
+				D_ASSERT(!parity_include);
+				ss = p * len;
+				p_offset = 0;
+				full = (1UL << p) - 1;
+
+			} else {
+				ss = k * len;
+				p_offset = p;
+			}
+
+ 			/* Walk from start to end by len, except for the last
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs